amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/hold/AddToHoldsBulkV1Tests.java

```diff
@@ -0,0 +1,519 @@
+            addedFiles.add(documentHeld);
+        }
+
+        RestRequestQueryModel queryReq = getContentFromSiteQuery(testSite.getId());
+        SearchRequest searchRequest = new SearchRequest();
+        searchRequest.setQuery(queryReq);
+
+        STEP("Wait until all files are searchable.");
+        await().atMost(30, TimeUnit.SECONDS)
+            .until(() -> getRestAPIFactory().getSearchAPI(null).search(searchRequest).getPagination()
+                .getTotalItems() == NUMBER_OF_FILES);
+
+        holdBulkOperation = HoldBulkOperation.builder()
+            .query(queryReq)
+            .op(HoldBulkOperationType.ADD).build();
+    }
+
+    /**
+     * Given a user with the add to hold capability and hold filling permission
+     * When the user adds content from a site to a hold using the bulk API
+     * Then the content is added to the hold and the status of the bulk operation is DONE
+     */
+    @Test
+    public void addContentFromTestSiteToHoldUsingBulkAPI()
+    {
+        UserModel userAddHoldPermission = roleService.createUserWithSiteRoleRMRoleAndPermission(testSite,
+            UserRole.SiteCollaborator, hold.getId(), UserRoles.ROLE_RM_MANAGER, PERMISSION_FILING);
+        users.add(userAddHoldPermission);
+
+        STEP("Add content from the site to the hold using the bulk API.");
+        HoldBulkOperationEntry bulkOperationEntry = getRestAPIFactory().getHoldsAPI(userAddHoldPermission)
+            .startBulkProcess(holdBulkOperation, hold.getId());
+
+        // Verify the status code
+        assertStatusCode(ACCEPTED);
+        assertEquals(NUMBER_OF_FILES, bulkOperationEntry.getTotalItems());
+
+        STEP("Wait until all files are added to the hold.");
+        await().atMost(20, TimeUnit.SECONDS).until(
+            () -> getRestAPIFactory().getHoldsAPI(getAdminUser()).getChildren(hold.getId()).getEntries().size()
+                == NUMBER_OF_FILES);
+        List<String> holdChildrenNodeRefs = getRestAPIFactory().getHoldsAPI(userAddHoldPermission)
+            .getChildren(hold.getId()).getEntries().stream().map(HoldChildEntry::getEntry).map(
+                HoldChild::getId).toList();
+        assertEquals(addedFiles.stream().map(FileModel::getNodeRefWithoutVersion).sorted().toList(),
+            holdChildrenNodeRefs.stream().sorted().toList());
+
+        STEP("Check the bulk status.");
+        HoldBulkStatus holdBulkStatus = getRestAPIFactory().getHoldsAPI(userAddHoldPermission)
+            .getBulkStatus(hold.getId(), bulkOperationEntry.getBulkStatusId());
+        assertBulkProcessStatus(holdBulkStatus, NUMBER_OF_FILES, 0, null);
+
+        STEP("Check the bulk statuses.");
+        HoldBulkStatusCollection holdBulkStatusCollection = getRestAPIFactory().getHoldsAPI(userAddHoldPermission)
+            .getBulkStatuses(hold.getId());
+        assertEquals(1, holdBulkStatusCollection.getEntries().size());
+        assertEquals(holdBulkStatus, holdBulkStatusCollection.getEntries().get(0).getEntry());
```

Comment:
Can't you test equality of the collection in a single call?

Reply:
Changed to have a single call.